app/lib/.server/llm/constants.ts

```diff
@@ -3,3 +3,36 @@ export const MAX_TOKENS = 8000;
 
 // limits the number of model responses that can be returned in a single request
 export const MAX_RESPONSE_SEGMENTS = 2;
+
+export interface File {
+  type: 'file';
+  content: string;
+  isBinary: boolean;
+}
+
+export interface Folder {
+  type: 'folder';
+}
+
+type Dirent = File | Folder;
+
+export type FileMap = Record<string, Dirent | undefined>;
+
+export const IGNORE_PATTERNS = [
```

Comment:
I really do thin we need a better/consistent way of maintaining exclusions. I wish we had a files-management api which provides relevant files (same for import/export)